src/ardrive.ts

```diff
@@ -341,6 +341,12 @@ export class ArDrive extends ArDriveAnonymous {
 			throw new Error(errorMessage.cannotMoveIntoSamePlace('Folder', newParentFolderId));
 		}
 
+		// Assert that there are no duplicate names in the destination folder
+		const entityNamesInParentFolder = await this.arFsDao.getPublicChildNamesOfParentFolderId(newParentFolderId);
```

Comment:
How about a simpler name? Maybe:
`getNamesOfPublicEntitiesWithParentFolderId`
or
`getPublicEntityNamesInFolder`

Reply:
Same comment applies to the private use case.